sample_2.py

```diff
@@ -0,0 +1,17 @@
+class Solution:
+   def fib(self, N: int) -> int:
+       if N <= 1:
+           return N
+       return self.memoize(N)
+
+   def memoize(self, N: int) -> {}:
+       cache = {0: 0, 1: 1}
```

Comment:
you can just replace it with simple variables-
a = 0
b = 1
c = None

Reply:
Using dictionary rather than variables makes the logic simpler, I think. Let's keep the dictionary indexing.

Reply:
You can add isinstance(n, int) here as well to check if number is intended or not

Reply:
You can use list also instead